AcceptanceTests/AcceptanceTests.PageObject/Helpers/PageNavigator.cs

```diff
@@ -11,39 +12,53 @@ public class PageNavigator
 
         public static void CompleteJourney(UserJourney userJourney)
         {
+            var targetPageType = userJourney.Pages.Last().GetType();
+
             foreach (var page in userJourney.Pages)
             {
                 CurrentPage = page;
+                var headingToPrint = string.IsNullOrEmpty(CurrentPage.HeadingText) ? CurrentPage.Uri : CurrentPage.HeadingText;
 
                 if (CurrentPage.GetType().IsEquivalentTo(typeof(DashboardPage)))
                 {
-                    switch (userJourney.DashboardOption)
-                    {
-                        case DashboardOption.BookAvideoHearing:
-                            var dashboardPage = (DashboardPage)CurrentPage;
-                            dashboardPage.BookHearing();
-                            break;
-                        default:
-                            throw new NotImplementedException($"Dashboard option {userJourney.DashboardOption} is not implemented.");
-                    }
-                } else
+                    CurrentPage.IsPageLoaded();
+                    ChooseDashboardFlow(userJourney.DashboardOption);
+                }
+                else if (CurrentPage.GetType().IsEquivalentTo(targetPageType))
+                {
+                    Console.WriteLine($"Reached target page {headingToPrint}");
+                    return;
+                }
+                else
                 {
-                    try
-                    {
-                        var headingToPrint = string.IsNullOrEmpty(CurrentPage.HeadingText) ? CurrentPage.Uri : CurrentPage.HeadingText;
+                    //try
+                    //{
                         Console.WriteLine($"Current page: {headingToPrint}");
                         CurrentPage.FillDetails(default);
                         CurrentPage.Continue();
-                    }
-                    catch (Exception ex)
-                    {
-                        Console.WriteLine($"Exception occurred {ex.Message}");
-                        Console.WriteLine("Page has no forms to be filled.");
-                        Console.WriteLine("Continuing to next page.");
-                        CurrentPage.Continue();
-                    }
+                    //}
```

Comment:
Is this commented out code still needed?

Reply:
Removed.